nipype/interfaces/freesurfer/longitudinal.py

```diff
@@ -251,3 +253,207 @@ def _list_outputs(self):
+    """Uses the longitudinal pipeline of recon-all to create a template for a given number of subject's sessions.
+
+    Examples
+    --------
+    >>> from nipype.interfaces.freesurfer.longitudinal import BaseReconAll
+    >>> baserecon = BaseReconAll()
+    >>> baserecon.inputs.base_id = 'sub-template'
+    >>> baserecon.inputs.timepoints = ['ses-1','ses-2']
+    >>> baserecon.inputs.directive = 'all'
+    >>> baserecon.inputs.subjects_dir = '.'
+    >>> baserecon.cmdline
+    'recon-all -all -base sub-template -sd . -tp ses-1 -tp ses-2'
+    """
+
+    _cmd = "recon-all"
+    input_spec = BaseReconAllInputSpec
+    output_spec = BaseReconAllOutputSpec
+    _can_resume = True
+    force_run = False
+
+    def _gen_subjects_dir(self):
+        return os.getcwd()
+
+
+    def _gen_filename(self, name):
+        if name == "subjects_dir":
+            return self._gen_subjects_dir()
+        return None
+
+
+    def _list_outputs(self):
+        if isdefined(self.inputs.subjects_dir):
+            subjects_dir = self.inputs.subjects_dir
+        else:
+            subjects_dir = self._gen_subjects_dir()
+
+        if isdefined(self.inputs.hemi):
+            hemi = self.inputs.hemi
+        else:
+            hemi = "both"
+
+        outputs = self._outputs().get()
+
+        outputs.update(
+            FreeSurferSource(subject_id=self.inputs.base_id,
+                subjects_dir=subjects_dir, hemi=hemi)._list_outputs()
+        )
+        outputs["subject_id"] = self.inputs.base_id
+        outputs["subjects_dir"] = subjects_dir
+        return outputs
+
+
+    def _is_resuming(self):
+        subjects_dir = self.inputs.subjects_dir
+        if not isdefined(subjects_dir):
+            subjects_dir = self._gen_subjects_dir()
+        if os.path.isdir(os.path.join(subjects_dir, self.inputs.base_id, "mri")):
+            return True
+        return False
+
+
+    def _format_arg(self, name, trait_spec, value):
+        return super(BaseReconAll, self)._format_arg(name, trait_spec, value)
+
+
+    @property
+    def cmdline(self):
+        cmd = super(BaseReconAll, self).cmdline
+
+        if not self._is_resuming():
+            return cmd
+
+        subjects_dir = self.inputs.subjects_dir
+        if not isdefined(subjects_dir):
+            subjects_dir = self._gen_subjects_dir()
+
+        directive = self.inputs.directive
+        if not isdefined(directive):
+            steps = []
+
+        iflogger.info(f"recon-all: {cmd}")
+        return cmd
+
+
+class LongReconAllInputSpec(ReconAllInputSpec):
+    subject_id = traits.Str(argstr="-subjid %s", desc="subject name")
+    long_id = traits.Str(
+        argstr="-long %s",
+        desc="longitudinal session/timepoint id",
+        xor=["subject_id"],
+        requires=["base_id"],
+        position=1
+    )
+    base_id = traits.Str(
+        argstr="%s",
+        desc="longitudinal base template id",
+        requires=["long_id"],
+        position=2
+    )
```

Comment:
Similar comment as for `BaseReconAll` regarding namings and constraints.

```suggestion
    longitudinal_timepoint_id = traits.Str(
        argstr="-long %s",
        desc="longitudinal session/timepoint id",
        mandatory=True
        position=1
    )
    longitudinal_template_id = traits.Str(
        argstr="%s",
        desc="longitudinal base template id",
        mandatory=True,
        position=2
    )
```